test/plugin/base_test.rb

```diff
@@ -1025,17 +1025,26 @@ def test_http_request_partial_from_record
     end
   end
 
-  def test_http_request_without_referer_from_record
+  def test_http_request_from_record_with_referer_nil_or_absent
     setup_gce_metadata_stubs
-    setup_logging_stubs do
-      d = create_driver
-      d.emit('httpRequest' => http_request_message_without_referer)
-      d.run
-    end
-    verify_log_entries(1, COMPUTE_PARAMS, 'httpRequest') do |entry|
-      assert_equal http_request_message_without_referer, entry['httpRequest'],
-                   entry
-      assert_nil get_fields(entry['structPayload'])['httpRequest'], entry
+    {
+      http_request_message_with_referer_nil => \
+        http_request_message_expected_struct_payload,
+      http_request_message_with_referer_absent => 'null'
+    }.each do |message, expected|
+      @logs_sent = []
+      setup_logging_stubs do
+        d = create_driver
+        d.emit('httpRequest' => message)
+        d.run
+      end
+      verify_log_entries(1, COMPUTE_PARAMS, 'httpRequest') do |entry|
+        assert_equal http_request_message_expected_without_referer,
+                     entry['httpRequest'], entry
+        assert_equal expected,
+                     get_fields(entry['structPayload'])['httpRequest'].to_json,
```

Comment:
Right now when we get a payload like:

```
{"httpRequest"=>{"referer"=>nil}}
```

The constructed proto is:

```
<Google::Protobuf::Struct: fields: {"httpRequest"=><Google::Protobuf::Value: null_value: :NULL_VALUE, number_value: 0.0, string_value: "", bool_value: false, struct_value: <Google::Protobuf::Struct: fields: {"referer"=><Google::Protobuf::Value: null_value: :NULL_VALUE, number_value: 0.0, string_value: "", bool_value: false, struct_value: nil, list_value: nil>}>, list_value: nil>}>
```

I found it weird at first sight too. Another option is to construct something like:
```
<Google::Protobuf::Struct: fields: {"httpRequest"=><Google::Protobuf::Value: null_value: :NULL_VALUE, number_value: 0.0, string_value: "", bool_value: false, struct_value: nil, list_value: nil>}>
```

But that means the following payloads would be equivalent:

```
{"httpRequest"=>{"referer"=>nil}}
{"httpRequest"=>nil}
```

Which is probably fine, I guess? But I did not see the necessity to add extra logic to behave it this way. Is there?